assets/js/components/surveys/CurrentSurvey.js

```diff
@@ -16,32 +16,173 @@
+		sendSurveyEvent( 'survey_closed' );
+
+		setValues( formName, { hideSurvey: true } );
+	}, [ formName, sendSurveyEvent, setValues ] );
+
+	const dismissSurvey = useCallback( () => {
+		sendSurveyEvent( 'survey_closed' );
 
-	if ( ! usingProxy || ! currentSurvey ) {
+		setValues( formName, { hideSurvey: true } );
+	}, [ formName, sendSurveyEvent, setValues ] );
+
+	// Check to see if a completion trigger has been met.
+	let triggeredCompletion;
+	// We only have trigger conditions for questions that are answered with
+	// ordinal values right now.
+	const ordinalAnswerMap = answers?.length ? answers.reduce( ( acc, answer ) => {
+		return {
+			...acc,
+			[ answer.question_ordinal ]: answer.answer_ordinal,
+		};
+	}, {} ) : [];
+
+	// Use Array.some to avoid looping through all completions; once a matching
+	// completion has been found, treat the survey as complete.
+	completions?.some( ( completion ) => {
+		completion.trigger_condition.some( ( condition ) => {
+			// If a question was answered with the appropriate value, a completion
+			// trigger has been fulfilled and we should treat this survey as complete.
+			if ( condition.answer_ordinal.includes( ordinalAnswerMap[ condition.question_ordinal ] ) ) { // eslint-disable-line camelcase
+				triggeredCompletion = completion;
+				return true;
+			}
+
+			return false;
+		} );
+
+		if ( triggeredCompletion ) {
+			return true;
+		}
+
+		return false;
+	} );
+
+	// If no specific trigger has been found but all questions are answered, use
+	// the first available trigger.
+	if ( ! triggeredCompletion && questions?.length && currentQuestionOrdinal > questions?.length ) {
+		triggeredCompletion = completions[ 0 ];
+	}
+
+	useEffect( () => {
+		if ( triggeredCompletion && ! hasSentCompletionEvent ) {
+			sendSurveyEvent( 'completion_shown', {
+				completion_ordinal: triggeredCompletion.completion_ordinal,
+			} );
+			setHasSentCompletionEvent( true );
+		}
+	}, [ hasSentCompletionEvent, sendSurveyEvent, triggeredCompletion ] );
+
+	if ( shouldHide || ! questions || ! completions ) {
 		return null;
 	}
 
-	return (
-		<Portal slug="survey">
-			<Survey
-				questions={ currentSurvey.question }
-				completions={ currentSurvey.completion }
+	if ( triggeredCompletion ) {
+		return (
+			<SurveyCompletion
+				dismissSurvey={ dismissSurvey }
+				ctaOnClick={ ctaOnClick }
+				ctaText={ triggeredCompletion.follow_up_text }
+				ctaURL={ triggeredCompletion.follow_up_url }
+				title={ triggeredCompletion.completion_title }
+			>
+				{ triggeredCompletion.completion_text }
+			</SurveyCompletion>
+		);
+	}
+
+	// eslint-disable-next-line camelcase
+	if ( currentQuestion?.question_type === 'rating' ) {
+		return (
+			<SurveyQuestionRating
+				answerQuestion={ answerQuestion }
+				choices={ currentQuestion.question.answer_choice }
+				dismissSurvey={ dismissSurvey }
+				key={ currentQuestion.question_ordinal }
+				question={ currentQuestion.question_text }
 			/>
-		</Portal>
-	);
+		);
```

Comment:
How about we extract a `[type]: Component` map here? I believe the props will always be the same to all question types so maybe the only condition here would be "does a component for this type exist" rather than checking the specific type.